android/java/org/chromium/chrome/browser/BraveRewardsNativeWorker.java

```diff
@@ -459,8 +462,11 @@ public void SetAutoContributionAmount(double amount) {
     }
 
     public void StartProcess() {
-        synchronized (lock) {
-            BraveRewardsNativeWorkerJni.get().startProcess(mNativeBraveRewardsNativeWorker);
+        if (ChromeFeatureList.isEnabled(BraveFeatureList.BRAVE_REWARDS)
+                && !BravePrefServiceBridge.getInstance().getSafetynetCheckFailed()) {
```

Comment:
This only checks flag state, but it doesn't start process of actual safetynet check. I suppose we should call `SafetyNetCheck.updateSafetynetStatus` here and only if it is successful `startProcess`.

Reply:
See `brave/android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentServiceImpl.java:58` for example